byzcoin/contracts.go

```diff
@@ -302,3 +274,54 @@ func (s *Service) ContractDarc(cdb CollectionView, inst Instruction, coins []Coi
 		return nil, nil, errors.New("unknown instruction type")
 	}
 }
+
+// loadConfigFromTrie loads the configuration data from the trie.
+func loadConfigFromTrie(st ReadOnlyStateTrie) (*ChainConfig, error) {
+	// Find the genesis-darc ID.
+	val, contract, _, err := getValueContract(st, NewInstanceID(nil).Slice())
```

Comment:
Could/should this be using VerifyAndDecode?

Reply:
i'm not sure why this would be `VerifyAndDecode`, which is a method of `Proof`. these functions are copy pasted from `struct.go` because they're mostly (only?) used for the contracts.